### ISSUE TITLE
Typescript typings for dynamic masks are incomplete

### ISSUE THREAD

**Issue by Shreya26042000** [uNmAnNeR/imaskjs#793]
<img width="954" alt="Screen Shot 2022-12-05 at 11 39 04 AM" src="https://user-images.githubusercontent.com/57302349/205704943-2d1e318d-22de-4b0a-9054-75a937c37fda.png">

`AnyMaskedOptions` does not support adding custom fields to mask, which should be supported from the second example in the docs. 

**To Reproduce**
Go to the project with TS, try to create a dynamic mask according to the example

**Environment**
imask 6.4.3

**Comment by adolfoscheidt** [comment]
Hello, I'm having a related issue with the `blocks` option while creating a Date mask:

``` 
const opts: IMask.MaskedDateOptions = {
    mask: Date,
    (...)
    blocks: {
      YYYY: {
        mask: IMask.MaskedRange,
        from: 1970,
        to: 2040,
      },
      A: {
        mask: IMask.MaskedEnum,
        enum: ['AM, PM'],
        // Type '{ mask: typeof IMask.MaskedEnum; enum: string[]; }' is not assignable to type 'AnyMaskedOptions'.
  Object literal may only specify known properties, and 'enum' does not exist in type 'AnyMaskedOptions'.
      },
    },
  };
```

After reading the [guide](https://imask.js.org/guide.html#masked-date) I believe it should be possible to create custom blocks like this `enum` I showed.
It seems the type `AnyMaskedOptions` does not include the type `MaskedEnumOptions` among its possible values, triggering this error. Is this correct? 

I'm using `react-imask 6.4.3`.
Thanks!

**Comment by mtlehtin** [comment]
An example grabbed from the guide page:
https://imask.js.org/guide.html#masked-pattern

```typescript
import type { AnyMaskedOptions } from 'imask'

const zipMask: AnyMaskedOptions = {
  mask: '#00000',
  definitions: {
    // <any single char>: <same type as mask (RegExp, Function, etc.)>
    // defaults are '0', 'a', '*'
    '#': /[1-6]/
  }
}
```

Throws `Type 'RegExp' is not assignable to type 'PatternDefinition'. ts(2322)`

**Comment by uNmAnNeR** [comment]
@mtlehtin @adolfoscheidt your cases are fixed in [3b3b6ae](https://github.com/uNmAnNeR/imaskjs/commit/3b3b6ae1952a4255abcb4719a45ac4ef5744a6d9)
To fix the original case need to do breaking change, so postpone for now.

**Comment by uNmAnNeR** [comment]
extra fields are not added but you can cast types with `as`

**Comment by IgorRio29** [comment]
Hello
I've started using v 7.1.3 and still see the original issue related to _startsWith_ property.
Can u please suggest how it can be resolved?
As per [guide ](https://imask.js.org/guide.html#masked-dynamic) seems it should work.

**Comment by uNmAnNeR** [comment]
should work out of the box in 7.2.0
